### PR TITLE
feat: added support for conditional parameters in hyperparameter tuning

### PR DESCRIPTION
**Pull request by rosiezou**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: tests/unit/aiplatform/test_hyperparameter_tuning_job.py [hunk 1]
```diff
@@ -377,6 +403,12 @@ def test_create_hyperparameter_tuning_job(
             base_output_dir=test_custom_job._TEST_BASE_OUTPUT_DIR,
         )
 
+        conditional_parameter_decay = hpt.DoubleParameterSpec(
```
Comment (jaycee-li) [inline tests/unit/aiplatform/test_hyperparameter_tuning_job.py:406]: These two are defined multiple times. We can declare them globally. 
E.g, `_TEST_CONDITIONAL_PARAMETER_DECAY = hpt.DoubleParameterSpec(...)`

**Review by jaycee-li — CHANGES_REQUESTED** [review]

File: google/cloud/aiplatform/hyperparameter_tuning.py [hunk 2]
```diff
@@ -105,6 +129,8 @@ def __init__(
         min: float,
         max: float,
         scale: str,
+        conditional_parameter_spec: Optional[Dict[str, "_ParameterSpec"]] = None,
```
[the remark this answers is not in the document]
Comment (rosiezou) [reply google/cloud/aiplatform/hyperparameter_tuning.py:132]: The only restriction is what the parent parameter can be -- looks like right now only `[IntegerParameterSpec, CategoricalParameterSpec, DiscreteParameterSpec]` are supported as parent parameters. But the conditional parameter itself can be any one of the four parameter specs. Additionally, defining this to take in an optional dictionary of the base class is consistent with what Sasha originally wrote in the base class constructor.

File: tests/unit/aiplatform/test_hyperparameter_tuning_job.py (hunk 1 above)
Comment (rosiezou) [reply tests/unit/aiplatform/test_hyperparameter_tuning_job.py]: added global definition

File: google/cloud/aiplatform/hyperparameter_tuning.py [hunk 3]
```diff
@@ -29,6 +29,10 @@
     "unspecified": gca_study_compat.StudySpec.ParameterSpec.ScaleType.SCALE_TYPE_UNSPECIFIED,
 }
 
+_INT_VALUE_SPEC = "integer_value_spec"
```
[the remark this answers is not in the document]
Comment (rosiezou) [reply google/cloud/aiplatform/hyperparameter_tuning.py:32]: updated

File: google/cloud/aiplatform/hyperparameter_tuning.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaycee-li) [reply google/cloud/aiplatform/hyperparameter_tuning.py:132]: That makes sense. Thanks for explaining!

**Review by jaycee-li — APPROVED** [review]

**Pull request merged**